lib/cap.c

```diff
@@ -472,10 +473,22 @@ caps_create(vfu_ctx_t *vfu_ctx, vfu_cap_t **vfu_caps, int nr_caps, int *err)
 }
 
 size_t
-vfu_pci_find_next_capability(vfu_ctx_t *vfu_ctx, size_t offset, int cap_id)
+vfu_pci_find_next_capability(vfu_ctx_t *vfu_ctx, bool extended,
+                             size_t offset, int cap_id)
 {
+    size_t space_size = vfu_ctx->reg_info[VFU_PCI_DEV_CFG_REGION_IDX].size;
     vfu_pci_config_space_t *config_space;
 
+    if (extended) {
+        errno = ENOTSUP;
+        return 0;
+    }
+
+    if (offset >= space_size) {
```

Comment:
We should also have a unit test checking for some common scenarios.

Reply:
indeed we should!